features/environment.py

```diff
@@ -45,7 +45,7 @@ def _is_accessible(self):
     def _start(self):
         """start process"""
 
-    def start(self, max_wait_limit=5):
+    def start(self, max_wait_limit=5, should_wait=True):
```

Comment:
Most of the changes to the `environment.py` are not needed:
1. `should_wait` argument isn't needed and in fact even bad. Starting Patroni process takes some time and it also doesn't fail instantly due to `ensure_unique_name()`. That is, we still have to wait in the below loop. But, in addition to `self._is_accessible()` also move `assert self._has_started(), "Process {0} is not running after being started".format(self._name)` to the loop. The exception raised from the loop should be intercepted/handled in `start_duplicate_patroni()` as expected one.
2. Refactoring around `__PG_PORT`,  `__BASE_API_PORT`, and `port_offset` is also not required. Instead you can simply pass `config = {'name': name, 'restapi': {'connect_address': 'localhost:8008'}}` from `start_duplicate_patroni()`. It will work because `self.recursive_update(config, custom_config)` is called after the main config handling in __make_patroni_test_config()` is performed.
3. the new `is_running()` method isn't used.